### PR TITLE
Doc: Cleanup RTD config

### PR DESCRIPTION
**Pull request by dweindl**
fixes various issues and enables fail-on-warning

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+version: 2
+
+sphinx:
+   configuration: doc/conf.py
+   fail_on_warning: true
+
+
+python:
+   version: 3.7
```
Comment (yannikschaelte) [inline .readthedocs.yaml:12]: 3.8/9 should also work

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+   install:
+   - requirements: .rtd_pip_reqs.txt
+   - method: pip
+     path: .
```
Comment (yannikschaelte) [inline .readthedocs.yaml:16]:
```suggestion
   - method: pip
     path: .
     extra_requirements:
     - doc
```
may be nicer

File: doc/editorial_board.rst [hunk 3]
```diff
@@ -1,28 +0,0 @@
-Editorial board
-===============
```
Comment (yannikschaelte) [inline doc/editorial_board.rst:2]: Why is it gone?

File: doc/conf.py [hunk 4]
```diff
@@ -94,3 +108,13 @@ def setup(app):
...
 html_logo = 'logo/PEtab.png'
+
+
+def skip_some_objects(app, what, name, obj, skip, options):
+    """Exclude some objects from the documentation"""
+    if getattr(obj, '__module__', None) == 'collections':
+        return True
+
+
+def setup(app):
```
Comment (yannikschaelte) [inline doc/conf.py:119]: docstring

**Review by yannikschaelte — APPROVED** [review]

File: doc/editorial_board.rst (hunk 3 above)
Comment (yannikschaelte) [reply doc/editorial_board.rst:2]: oh, I see, it's only the library.

File: .readthedocs.yaml (hunk 1 above)
Comment (dweindl) [reply .readthedocs.yaml]: > Problem in your project's configuration. Invalid "python.version": expected one of (2, 2.7, 3, 3.5, 3.6, 3.7, 3.8, pypy3.5), got 3.9

almost

**Pull request merged**